src/page-modules/contact/tickets-app/tickets-app-form-machine.ts

```diff
@@ -0,0 +1,388 @@
+  lastName?: string;
+  email?: string;
+  phoneNumber?: string;
+  address?: string;
+  postalCode?: string;
+  city?: string;
+  bankAccountNumber?: string;
+  IBAN?: string;
+  SWIFT?: string;
+  question?: string;
+  orderId?: string;
+  customerNumber?: string;
+  travelCardNumber?: string;
+  refundReason?: string;
+  amount?: string;
+};
+
+export type ContextProps = {
+  formType?: FormType;
+  attachments?: File[];
+  firstName?: string;
+  lastName?: string;
+  email?: string;
+  phoneNumber?: string;
+  address?: string;
+  postalCode?: string;
+  city?: string;
+  hasInternationalBankAccount: boolean;
+  bankAccountNumber?: string;
+  IBAN?: string;
+  SWIFT?: string;
+  question?: string;
+  orderId?: string;
+  customerNumber?: string;
+  travelCardNumber?: string;
+  refundReason?: string;
+  amount?: string;
+  errorMessages: InputErrorMessages;
+};
+
+const setInputToValidate = (context: ContextProps) => {
+  // Destructure the needed fields from context
+  const {
+    formType,
+    attachments,
+    firstName,
+    lastName,
+    email,
+    phoneNumber,
+    address,
+    postalCode,
+    city,
+    hasInternationalBankAccount,
+    bankAccountNumber,
+    IBAN,
+    SWIFT,
+    question,
+    orderId,
+    customerNumber,
+    travelCardNumber,
+    refundReason,
+    amount,
+  } = context;
+
+  const commonAppFields = {
+    formType,
+    question,
+    attachments,
+    firstName,
+    lastName,
+    email,
+  };
+
+  switch (formType) {
+    case FormType.PriceAndTicketTypes:
+      return {
+        formType,
+        question,
+        firstName,
+        lastName,
+        email,
+      };
+
+    case FormType.AppTicketing:
+      return {
+        ...commonAppFields,
+        orderId,
+        phoneNumber,
+      };
+
+    case FormType.AppTravelSuggestion:
+      return {
+        ...commonAppFields,
+      };
+
+    case FormType.AppAccount:
+      return {
+        ...commonAppFields,
+        customerNumber,
+        phoneNumber,
+      };
+
+    case FormType.TravelCardQuestion:
+      return {
+        formType,
+        travelCardNumber,
+        question,
+        firstName,
+        lastName,
+        email,
+        phoneNumber,
+      };
+
+    case FormType.OrderTravelCard:
+      return {
+        formType,
+        firstName,
+        lastName,
+        address,
+        postalCode,
+        city,
+        email,
+      };
+
+    case FormType.AppTicketRefund:
+      return {
+        formType,
+        customerNumber,
+        orderId,
+      };
+
+    case FormType.OtherTicketRefund:
+      return {
+        formType,
+        travelCardNumber,
+        refundReason,
+        amount,
+        firstName,
+        lastName,
+        address,
+        postalCode,
+        city,
+        email,
+        phoneNumber,
+        ...(hasInternationalBankAccount
+          ? { IBAN, SWIFT }
+          : { bankAccountNumber }),
+      };
+  }
+};
+
+export const ticketsAppFormMachine = setup({
```

Comment:
Maybe we should name this "TicketingForm" instead? Would need to update a lot of places though

Reply:
Renamed from `tickets-app` to `ticketing`.